### DISCUSSION TITLE
transmission.log is a very big in size

### DISCUSSION THREAD

**Discussion by fdfytr** [haugene/docker-transmission-openvpn#2265] [Q&A] [answered]
It is over 1gb now
Is there any way to control the size, autodeletion etc.

**Comment by edgd1er** [comment, marked as answer]
logrotate is your friend.
to be configured on the host, not sure the package is installed in the container.

**Reply by fdfytr** [reply]
I wonder also why it happened in the first place since there is 10m limit in env at docker

**Reply by edgd1er** [reply]
if you refer to "--log-opt max-size=10m", this settings is for docker daemon limiting the container logs storage, ie logs sent from stderr/stdout/ Transmission is writing to /config/transmission-home/ folder.

**Comment by miaLDN** [comment]
Any chance of logrotate being added to the container please?  I have to cp /dev/null to transmisson.log whenever it reaches a few GB.

E2A:  I’d like to keep the log maintenance (logrotate etc.) compartmentalized inside the same container as the processes that are creating it.
The container may then run on any host, without external host dependencies.

**Comment by milkas9** [comment]
I had the same problem, my transmission log grew to 1.3GB+ without realizing. I have a script to rotate it now when it grows past 100MB. Runs weekly with cron.

```
#!/bin/bash

# Define the name of the Docker container and log file
DOCKER_CONTAINER_NAME="Transmission_VPN"
LOG_FILE="<your-path-here>/transmission.log"

# Stop the Docker container
echo "Stopping Docker container: $DOCKER_CONTAINER_NAME"
docker stop $DOCKER_CONTAINER_NAME

# Check if the log file exists and is larger than 100MB
if [ -f "$LOG_FILE" ]; then
    LOG_SIZE=$(du -m "$LOG_FILE" | cut -f1)
    if [ $LOG_SIZE -gt 100 ]; then
        # Rotate the log file
        echo "Rotating log file: $LOG_FILE"
        mv "$LOG_FILE" "$LOG_FILE.$(date +"%Y%m%d%H%M%S")"
        touch "$LOG_FILE"
        echo "Log file rotated."
    fi

    # Keep only the last 5 log files including the current one
    echo "Keeping only the last 5 log files"
    ls -1 -t "$LOG_FILE".* | tail -n +6 | xargs rm -f
else
    echo "Log file not found: $LOG_FILE"
fi

# Start the Docker container again
echo "Starting Docker container: $DOCKER_CONTAINER_NAME"
docker start $DOCKER_CONTAINER_NAME


```

**Reply by miaLDN** [reply]
Thanks for the reminder.  :)
Just checked, and my transmission.log is over 1GB again :(
